symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/datafeed/impl/DatafeedServiceV1.java

```diff
@@ -0,0 +1,148 @@
+        } while (this.started.get());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void stop() {
+        this.started.set(false);
+        this.datafeedId = null;
+    }
+
+    private void readDatafeed() throws Throwable {
+        this.retry.executeCheckedSupplier(() -> {
+            try {
+                List<V4Event> events = datafeedApi.v4DatafeedIdReadGet(datafeedId, authSession.getSessionToken(), authSession.getKeyManagerToken(), null);
+                if (events != null && !events.isEmpty()) {
+                    handleV4EventList(events);
+                }
+            } catch (ApiException e) {
+                log.error("Error {}: {}", e.getCode(), e.getMessage());
+                if (e.isUnauthorized()) {
+                    log.info("Re-authenticate and try again");
+                    try {
+                        authSession.refresh();
+                    } catch (AuthUnauthorizedException authUnauthorizedException) {
+                        log.error("Cannot re-authenticate the bot");
+                        throw authUnauthorizedException;
+                    }
+                }
+                if (e.isClientError()) {
+                    log.info("Recreate a new datafeed and try again");
+                    datafeedId = this.createDatafeedAndSaveToDisk();
+                }
+                throw e;
+            }
+            return null;
+        });
+
+    }
+
+    private String createDatafeedAndSaveToDisk() throws Throwable {
+        return this.retry.executeCheckedSupplier(() -> {
+            try {
+                String id = this.datafeedApi.v4DatafeedCreatePost(authSession.getSessionToken(), authSession.getKeyManagerToken()).getId();
+                this.writeDatafeedIdToDisk(id);
+                log.debug("Datafeed: {} was created and saved to disk", id);
+                return id;
+            } catch (ApiException e) {
+                log.error("Error {}: {}", e.getCode(), e.getMessage());
+                if (e.isUnauthorized()) {
+                    try {
+                        authSession.refresh();
+                    } catch (AuthUnauthorizedException authUnauthorizedException) {
+                        log.error("Cannot re-authenticate the bot");
+                        throw authUnauthorizedException;
+                    }
+                }
+                throw e;
+            }
+        });
+    }
+
+    private String retrieveDatafeedIdFromDisk() {
+        String datafeedId = null;
+        try {
+            File file = this.getDatafeedIdFile();
+            Path datafeedIdPath = Paths.get(file.getPath());
+            String[] persistedDatafeed = Files.readAllLines(datafeedIdPath).get(0).split("@");
+            datafeedId = persistedDatafeed[0];
+            log.info("Retrieve datafeed id from persisted file: {}", datafeedId);
+        } catch (IOException e) {
+            log.info("No datafeed id is persisted");
```

Comment:
IMHO this log is quite confusing, maybe we should say something like : `No persisted datafeed id could be retrieved from the filesystem`.
Maybe we would like to log the exception message `e.getMessage()` (which could be file not found, for instance)